### ISSUE TITLE
Create missing default

### ISSUE THREAD

**Issue by wenerme** [sinclairzx81/typebox#726]
```ts
test('missing defaults', () => {

  const T = Type.Object({
    a: Type.Object({
      b: Type.Array(Type.String(), { default: [] }),
    }, { default: {} }), // really hope typebox freeze/clone the default when returned by Value.Default
  });

  console.log(Value.Create(T)); // missing b
  console.log(Value.Default(T, {})); // !!! should not mutate the object, 
  console.log(structuredClone(Value.Default(T, {}))); // I expected the Create behave like this
});
```

**Comment by sinclairzx81** [comment]
@wenerme Hi,

The `Create` and `Default` functions have different semantics when instancing values.

Create
- Creates values derived from `type` information. 
- Will `throw` if value cannot be instanced.
- Schema `default` annotations are optional
- Returns a static type

Default
- Patches values derived from `default` annotations only
- Will not `throw` if value cannot be instanced.
- Schema `default` annotations are required
- Returns unknown

These functions are the inverse of each other and are used different ways. The Create function is used when you want a NEW value instanced. The Default function is used when you want to PATCH an existing value for missing properties (usually used prior to a validation check to instance sub properties of that value).

Mutability

The `Default`, `Clean` and `Mutate` functions are all documented as mutable functions (as well as commented as `[Mutable]`). This is intentional and primarily done to boost performance in validation pipelines and avoid unnecessary instancing of JavaScript objects (which is very slow). From the readme.

> Use Default to generate missing properties on a value using default schema annotations if available. This function does not check the value and returns an unknown type. You should Check the result before use. *Default is a mutable operation. To avoid mutation, Clone the value first.*

So, to avoid mutation, you should clone the value (which is a opt-in performance cost)

```typescript
const D = Value.Default(T, Value.Clone({})) // To avoid mutation, clone the value.
```
Hope this helps
S

**Comment by wenerme** [comment]
for `console.log(Value.Default(T, {}));` I mean

```ts
  const T = Type.Object({
    a: Type.Object({
      b: Type.Array(Type.String(), { default: [] }),
    }, { default: {} }),
  });
  (Value.Default(T, {}) as any).a.b.push('1')
  console.log(Value.Default(T, {})); // the default on schema changed
```


I use zod before, so I try to find `parse` in typebox, zod parse do not mutate(that's also a problem, somthing want to mutate), so I think muatte `Value.Default(T, {})` result is ok, which is not.

---

BTW, I also want to a easier way to parse existing, is this right ?

```ts
console.log(
    // get T, run transform
    Value.Decode(T,
      // remove additional
      Value.Clean(T,
        // add missing
        Value.Default(T,
          // '1' -> 1
          Value.Convert(T, {
            z: 1, n: '1', d: new Date(),
          }),
        ),
      ),
    ),
  );
```

**Comment by sinclairzx81** [comment]
@wenerme Hiya,

> the default on schema changed

Ah yes, I see it. This is indeed a bug (as schematics are treated as immutable). Will look at fixing this over the next few days. Good find!

> BTW, I also want to a easier way to parse existing, is this right ?

Yes, this code is correct. Combined these functions and would constitute a `Parse` operation.

```typescript
const Parse = <T extends TSchema>(T: T, value: unknown) => 
  Value.Decode(T,
    Value.Clean(T,
      Value.Default(T,
        Value.Convert(T, value),
      )
    )
  )

const value = Parse(Type.String(), 'hello world')
```

**Comment by sinclairzx81** [comment]
@wenerme Heya,

Have quickly patched this on 0.32.7. If you want to give this a quick test your side, can look to close off this issue.

Thanks for reporting this, it's a excellent catch!

Cheers

**Comment by sinclairzx81** [comment]
Actually, I'm going to take a look at Create also and ensure the default annotation is cloned on assignment there also.
